x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/ranges/ranges.tsx

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiForm } from '@elastic/eui';
+
+import {
+  autoInterval,
+  RangeType,
+  UI_SETTINGS,
+} from '../../../../../../../../src/plugins/data/common';
+import { RangeEditor } from './range_editor';
+import { OperationDefinition } from '../index';
+import { FieldBasedIndexPatternColumn } from '../column_types';
+import { updateColumnParam, changeColumn } from '../../../state_helpers';
+
+export const DEFAULT_INTERVAL = 1000;
+
+export const MODES = {
+  Range: 'range',
+  Histogram: 'histogram',
+} as const;
+
+export type RangeTypeLens = RangeType & { label: string };
+
+export type MODES_TYPES = typeof MODES.Range | typeof MODES.Histogram;
+
+export interface RangeIndexPatternColumn extends FieldBasedIndexPatternColumn {
+  operationType: 'range';
+  params: {
+    type: MODES_TYPES;
+    interval: 'auto' | number;
+    maxBars: 'auto' | number;
+    ranges: RangeTypeLens[];
+  };
+}
+
+export type RangeColumnParams = RangeIndexPatternColumn['params'];
+export type UpdateParamsFnType = <K extends keyof RangeColumnParams>(
+  paramName: K,
+  value: RangeColumnParams[K]
+) => void;
+
+export const isValidNumber = (value: number | '') =>
+  value !== '' && !isNaN(value) && isFinite(value);
+export const isRangeWithin = (range: RangeTypeLens): boolean => range.from <= range.to;
+const isFullRange = ({ from, to }: RangeType) => isValidNumber(from) && isValidNumber(to);
+export const isValidRange = (range: RangeTypeLens): boolean => {
+  const { from, to } = range;
+  return (isFullRange(range) && isRangeWithin(range)) || isValidNumber(from) || isValidNumber(to);
+};
```

Comment:
Something about the range validation isn't working to verify that the "to" value is higher than "from"

<img width="475" alt="Screen Shot 2020-09-09 at 6 53 40 PM" src="https://user-images.githubusercontent.com/666475/92663615-be1dd680-f2cf-11ea-8e68-4526241f9ffe.png">